pkg/server/event_handler.go

```diff
@@ -228,9 +234,18 @@ func (h *EventHandler[T, P]) Handle(ctx context.Context, m pubsub.Message) error
 	if err != nil {
 		return fmt.Errorf("failed to convert object to pmap event payload: %w", err)
 	}
-	// TODO(#21): Add additional metadata to pmap event.
+
+	var gr *v1alpha1.GitHubSource
+	if m.Attributes["payloadFormat"] == "JSON_API_V1" {
```

Comment:
What if it's not `JSON_API_V1`?

Reply:
If it's not `JSON_API_V1`, the payload would be `None`. In this case we don't need to parseGithubSource.
https://cloud.google.com/storage/docs/pubsub-notifications#payload